lib/pbench/server/api/resources/upload_api.py

```diff
@@ -103,61 +102,39 @@ def put(self, filename: str):
             )
             abort(HTTPStatus.BAD_REQUEST, message=msg)
 
+        msg = ""
+        status = None
         content_length_hdr = request.headers.get("Content-Length")
         if not content_length_hdr:
             msg = "Missing required Content-Length header"
+            status = HTTPStatus.LENGTH_REQUIRED
+        else:
+            try:
+                content_length = int(content_length_hdr)
+            except ValueError:
+                msg = f"Invalid Content-Length header, not an integer ({content_length_hdr})"
+                status = HTTPStatus.BAD_REQUEST
+            else:
+                if content_length > self.max_content_length:
+                    cl = humanize.naturalsize(content_length)
+                    mcl = humanize.naturalsize(self.max_content_length)
+                    msg = f"Content-Length greater than maximum allowed size ({cl} > {mcl})"
+                    status = HTTPStatus.REQUEST_ENTITY_TOO_LARGE
+                elif content_length == 0:
+                    msg = "Invalid Content-Length header, zero (0)"
+                    status = HTTPStatus.BAD_REQUEST
+                elif content_length < 0:
+                    msg = f"Invalid Content-Length header, negative ({content_length})"
+                    status = HTTPStatus.BAD_REQUEST
+        if msg:
```

Comment:
This is great, although if I'd coded it I think I'd've been more likely to use `if status:` ... both because that's the most important of the two, and because `None` is more "instantly intuitively self-obviously" falsey than `""`.